hibernate-core/src/main/java/org/hibernate/mapping/Component.java

```diff
@@ -172,9 +199,13 @@ public void addColumn(Column column) {
 	@Override
 	public List<Selectable> getSelectables() {
 		if ( cachedSelectables == null ) {
-			cachedSelectables = properties.stream()
+			final List<Selectable> selectables = properties.stream()
 					.flatMap( p -> p.getSelectables().stream() )
 					.collect( toList() );
+			if ( discriminator != null ) {
+				selectables.addAll( discriminator.getSelectables() );
```

Comment:
Adding the discriminator to the selectables definitevly makes things easier, though I would like to discuss the implications of that. For entities, we do not treat the discriminator to be part of the entity state, because that is encoded in the entity key through the entity name.
I understand that embeddable data does not have such a key where the type/discriminator is stored, so this might very well be the only option to model this. Let's have a chat about it anyway :)

Reply:
IIRC this might have only been needed to pass validation against the component's columns. Maybe I'm wrong, but if that's the case we might treat it specially there.